### PR TITLE
V3.7 optimize asset editing on multiple and undo

### PR DESCRIPTION
**Pull request by arsen2010**
Re: #

### Changelog

*

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

File: editor/inspector/assets/animation-graph.js [hunk 1]
```diff
@@ -36,12 +50,13 @@ exports.update = function(assetList, metaList) {
     this.meta = this.metaList[0];
     this.asset = this.assetList[0];
 
-    if (assetList.length !== 1) {
-        this.$.button.disabled = true;
-        this.$.tip.style.display = 'block';
+    if (assetList.length > 1) {
```
Comment (dogodo-cc) [inline editor/inspector/assets/animation-graph.js:53]: 此处没有容错，不知道是否有影响

File: editor/inspector/assets/fbx/index.js [hunk 2]
```diff
@@ -89,6 +86,9 @@ const Elements = {
             panel.$.tabPanel.addEventListener('change', () => {
                 panel.dispatch('change');
             });
+            panel.$.tabPanel.addEventListener('snapshot', () => {
+                panel.dispatch('snapshot');
```
Comment (dogodo-cc) [inline editor/inspector/assets/fbx/index.js:90]: 这样写是因为 snapshot 不会冒泡吗

Comment (arsen2010) [reply editor/inspector/assets/fbx/index.js:90]: 是的，ui-panel 里的 shadowDom 里的事件，不会冒泡到 ui-panel （父级）这一层上来

File: editor/inspector/assets/animation-graph.js (hunk 1 above)
Comment (arsen2010) [reply editor/inspector/assets/animation-graph.js:53]: 这是入参，默认是空数组

File: editor/inspector/contributions/asset.js [hunk 3]
```diff
@@ -326,10 +322,10 @@ const Elements = {
                 }
 
                 contentRender.__panels__ = Array.from(contentRender.children);
-                Array.prototype.forEach.call(contentRender.__panels__, ($panel) => {
+                for (const $panel of contentRender.__panels__) {
```
Comment (VisualSJ) [inline editor/inspector/contributions/asset.js]: 是不是应该使用 promise.all

File: editor/inspector/contributions/asset.js [hunk 4]
```diff
@@ -305,12 +305,8 @@ const Elements = {
...
                         contentRender.__panels__[i] = document.createElement('ui-panel');
-                        contentRender.__panels__[i].addEventListener('change', (event) => {
+                        contentRender.__panels__[i].addEventListener('change', () => {
                             Elements.header.isDirty.call(panel);
-
-                            if (!event || !event.args || !event.args[0] || event.args[0].snapshot !== false) {
-                                panel.history && panel.history.snapshot(panel);
-                            }
                         });
                         contentRender.__panels__[i].addEventListener('snapshot', () => {
```
Comment (VisualSJ) [inline editor/inspector/contributions/asset.js:311]: 这里应该是 confirm，否则大家在书写 asset 模版的时候又要多一个需要理解的事件

File: editor/inspector/assets/fbx/index.js (hunk 2 above)
Comment (doommm) [reply editor/inspector/assets/fbx/index.js:90]: https://developer.mozilla.org/zh-CN/docs/Web/API/Event/Event

使用参考文档里提到的 `composed` 参数，可以使事件支持冒泡，就不需要再这样处理了

Comment (arsen2010) [reply editor/inspector/assets/fbx/index.js:90]: uiPanel.dispatch 内部默认阻断，事件不一直冒泡；涉及面比较大，另议。

File: editor/inspector/contributions/asset.js (hunk 4 above)
Comment (arsen2010) [reply editor/inspector/contributions/asset.js:311]: snapshot 是业务特性，开发的时候需要这个特性的时候需要知道

File: editor/inspector/assets/fbx/index.js (hunk 2 above)
Comment (doommm) [reply editor/inspector/assets/fbx/index.js:90]: image
ui-panel 的 dispatch 实现后续需要调整

**Review by doommm — APPROVED** [review]

File: editor/inspector/contributions/asset.js (hunk 3 above)
Comment (arsen2010) [reply editor/inspector/contributions/asset.js]: 已调整为用 promise.all

**Comment by arsen2010** [comment]
@VisualSJ 请审核和合并

**Review by VisualSJ — APPROVED** [review]

File: editor/inspector/contributions/asset.js (hunk 4 above)
Comment (VisualSJ) [reply editor/inspector/contributions/asset.js:311]: undo 系统之后不会有 snapshot

**Review by dogodo-cc — APPROVED** [review]

**Pull request merged**